File: backend/internal/models/topics.go
package models

import "time"

type Topic struct {
    ID          uint      `gorm:"primaryKey" json:"id"`
    TopicName        string    `gorm:"not null" json:"topic_name"`
    Description string    `gorm:"not null" json:"description,omitempty"`
    TopicClass   string    `gorm:"not null" json:"topic_class"`
    
    // User details
    CreatedBy   uint      `json:"createdBy"`
    
    PostCount   int       `gorm:"default:0;not null" json:"post_count"`
    FollowerCount int       `gorm:"default:0;not null" json:"follower_count"`
    
    CreatedAt   time.Time `json:"createdAt"`
    UpdatedAt   time.Time `json:"updated_at"`
}


func (Topic) TableName() string {
	return "topics"
}

File: frontend/src/types/topics.ts
export interface Topic {
    id: number;
    topic_name: string;
    description?: string;
    topic_class: string;

    // user details
    createdBy: number;
    username?: string;

    user_has_followed: boolean;

    post_count: number;
    follower_count: number;
    followers: string[];

    createdAt: Date;
    updatedAt: Date;
}